components/DateSelectField.tsx

```diff
@@ -178,12 +178,12 @@ const DateSelectField: FC<DateSelectFieldProps> = ({
           message={errorMessage}
         />
       )}
-      <div className="flex flex-col space-y-2 sm:flex-row sm:space-x-2 sm:space-y-0">
+      <fieldset className="flex flex-col space-y-2 sm:flex-row sm:space-x-2 sm:space-y-0">
         <DateSelect
           ariaDescribedby={getAriaDescribedby()}
           dateSelectLabelId={dateSelectLabelId}
           error={!!errorMessage}
-          id={id}
+          id={`${id}-year`}
```

Comment:
This was done on pursose otherwise the error message link doesn't work. A11y shouldn't care about element's id.
`http://localhost:3000/en/status#dateOfBirth`

Reply:
If we want to keep the `-year` suffix then we need to modify how we build the error links.

Reply:
I've added the `mapping` code when building the error summry items. It fix the mentionned issue.

```tsx
getErrorSummaryItems(formikErrors, t).map((item) => {
    if (item.feildId !== 'dateOfBirth') return item
    // field id should target the year select input
    return { ...item, feildId: 'dateOfBirth-year' }
}),
```